src/types.ts

```diff
@@ -453,15 +457,22 @@ export interface AccountTransactionHeader {
     expiry: TransactionExpiry;
 }
 
-export interface SimpleTransfer {
+export interface SimpleTransferPayload {
     /** µGTU amount to transfer */
     amount: GtuAmount;
 
-    /** the recipient of the transfer*/
+    /** the recipient of the transfer */
     toAddress: AccountAddress;
 }
 
-export type AccountTransactionPayload = SimpleTransfer;
+export interface SimpleTransferWithMemoPayload extends SimpleTransferPayload {
+    /** The bytes representation of the memo of the transaction  */
+    memo: Buffer;
```

Comment:
To stay consistent with what we have done for other fields, we should introduce a `Memo` class that validates that the memo has a valid size. With the current implementation I won't know that my transaction is invalid until I send it to a node.

Reply:
Added such a class